File: app/src/main/java/org/cuieney/videolife/ui/adapter/MusicAdapter.java
package org.cuieney.videolife.ui.adapter;

import android.content.Context;
import android.support.v4.view.ViewCompat;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.ms.square.android.expandabletextview.ExpandableTextView;

import org.cuieney.videolife.R;
import org.cuieney.videolife.common.base.BaseRecycerViewAdapter;
import org.cuieney.videolife.common.image.ImageLoader;
import org.cuieney.videolife.entity.MusicListBean;

import java.util.List;

/**
 * Created by cuieney on 17/3/4.
 */

public class MusicAdapter extends BaseRecycerViewAdapter<MusicListBean, RecyclerView.ViewHolder> {
    public MusicAdapter(Context context, List<MusicListBean> list) {
        super(context, list);
    }

    @Override
    public RecyclerView.ViewHolder getCreateViewHolder(ViewGroup parent, int viewType) {
        if (viewType == R.layout.music_item) {
            return new MusicAdapter.MyViewHoler(inflater.inflate(viewType, parent, false));
        } else {

            return new MusicAdapter.TopHolder(inflater.inflate(viewType, parent, false));
        }
    }

    @Override
    public void getBindViewHolder(RecyclerView.ViewHolder holder, int position) {
        if (holder instanceof MyViewHoler) {
            MyViewHoler viewHoler = (MyViewHoler) holder;
            MusicListBean musicListBean = list.get(position);
            ViewCompat.setTransitionName(viewHoler.imageView, String.valueOf(position) + "_image");
            ImageLoader.loadAll(context, musicListBean.getOphoto(), viewHoler.imageView);
            viewHoler.title.setText(musicListBean.getMname());
            viewHoler.description.setText(musicListBean.getMdesc());
            viewHoler.itemView.setOnClickListener(v -> {
                if (mClickListener != null) {
                    mClickListener.onItemClick(position,v,holder);
                }
            });
        }

    }

    @Override
    public int getItemViewType(int position) {

        if (position == 0 || position == 1) {
            return R.layout.top_item;
        }
        return R.layout.music_item;

    }

    public static class MyViewHoler extends RecyclerView.ViewHolder {
        public TextView title;
        public TextView description;
        public ImageView imageView;
        public ExpandableTextView expandedDesc;

        public MyViewHoler(View itemView) {
            super(itemView);
            title = ((TextView) itemView.findViewById(R.id.title));
            description = ((TextView) itemView.findViewById(R.id.description));
            imageView = ((ImageView) itemView.findViewById(R.id.img));
        }
    }

    public static class TopHolder extends RecyclerView.ViewHolder {
        private TextView textView;

        public TopHolder(View itemView) {
            super(itemView);
            textView = ((TextView) itemView.findViewById(R.id.name));
        }
    }
}

File: app/src/main/java/org/cuieney/videolife/kotlin/ui/adapter/MusicAdapter.kt
package org.cuieney.videolife.kotlin.ui.adapter

import android.support.v4.view.ViewCompat
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import org.cuieney.videolife.R
import org.cuieney.videolife.common.base.BaseRecycerViewAdapter
import org.cuieney.videolife.common.image.ImageLoader
import org.cuieney.videolife.entity.MusicListBean
import kotlinx.android.synthetic.main.music_item.view.*
/**
 * Created by cuieney on 2017/6/5.
 */
class MusicAdapter(t:ArrayList<MusicListBean>):BaseAdapter<MusicListBean>(t){

    override fun getCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        if (viewType == R.layout.music_item) {
            return MyViewHoler(LayoutInflater.from(parent.context).inflate(viewType, parent, false))
        } else {

            return TopHolder(LayoutInflater.from(parent.context).inflate(viewType, parent, false))
        }
    }

    override fun getBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        if (holder is MyViewHoler) {
            holder.bindData(items[position],position)
            holder.itemView.setOnClickListener { v ->
                clickListener?.onItemClick(position, v, holder)
            }
        }
    }

    override fun getItemViewType(position: Int): Int {

        if (position == 0 || position == 1) {
            return R.layout.top_item
        }
        return R.layout.music_item

    }

    class MyViewHoler(itemView: View) : RecyclerView.ViewHolder(itemView) {

        fun bindData(data:MusicListBean,position: Int){
            ViewCompat.setTransitionName(itemView.img, position.toString() + "_image")
            ImageLoader.loadAll(itemView.context, data.ophoto, itemView.img)
            itemView.title.setText(data.mname)
            itemView.description.setText(data.mdesc)

        }

    }

    class TopHolder(itemView: View) : RecyclerView.ViewHolder(itemView)
}